### PR TITLE
Add required packages for recommendations engine

### LINKED ISSUE #105: Add support for ML recommender
It would be great to add a container setup to run the ML recommender system used by Totara Engage. System requirements and installation instructions here:

https://help.totaralearning.com/display/DEV/Recommender+installation+and+configuration

### PR DESCRIPTION
**Pull request by derschatta**
this fixes #105

### CODE REVIEW

**Comment by markmetcalfe** [comment]
Does it potentially make more sense to have a separate recommendations/python container?

**Comment by derschatta** [comment]
> Does it potentially make more sense to have a separate recommendations/python container?

thought about it but it also needs PHP. So it would make most sense to install it in the existing ones then you wouldn't need to create a container for every PHP version.

**Comment by elkbullwinkle** [comment]
It doesn't need php. It works in the way:
You run a scheduled task to run export (there is a separate cli file for it)
Then you run a python recommenders which should have access to the shared data directory
Then you run a scheduled task to do the import (there is a separate cli file for it).

Also you added dependencies to a docker file, the dependencies might change with a new version and might be generally Totara version specific.

**Comment by derschatta** [comment]
> It doesn't need php. It works in the way:
> You run a scheduled task to run export (there is a separate cli file for it)
> Then you run a python recommenders which should have access to the shared data directory
> Then you run a scheduled task to do the import (there is a separate cli file for it).

well, without PHP it wouldn't work. without it you can't run the export and import, can you?

> Also you added dependencies to a docker file, the dependencies might change with a new version and might be generally Totara version specific.

that's right but how would I solve this? The version you use is not connected to the containers. Only to the version you check out and use.

**Comment by derschatta** [comment]
I followed the instructions on https://help.totaralearning.com/display/DEV/Recommender+installation+and+configuration

> This option increases complexity, so instead of separate hosts consider installation full Totara on ML instance and connecting it to a read-only database slave, which will not be available via load balancer and only serve as a ML processor. Use the method below only when ML server must not have access to Totara database or must not have Totara instance installed.

For me having a separate container would mean overcomplicating the setup.

File: php/php72/Dockerfile [hunk 1]
```diff
@@ -114,3 +105,38 @@ RUN git clone https://github.com/zsh-users/zsh-syntax-highlighting ~/.oh-my-zsh/
...
+    urllib3==1.25.9
+
+RUN ln -fs /usr/share/zoneinfo/${TIME_ZONE} /etc/localtime \
+    && dpkg-reconfigure --frontend noninteractive tzdata
+
+COPY config/php.ini /usr/local/etc/php/
+COPY config/fpm.conf /usr/local/etc/php-fpm.d/zz-totara.conf
+
+# Source each .sh file found in the /shell/ folder
+RUN echo 'for f in ~/custom_shell/*.sh; do [[ -e "$f" ]] && source "$f"; done;' >> ~/.bashrc
```
Comment (markmetcalfe) [inline php/php72/Dockerfile:142]: This line (loading the aliases into the container) needs to go after the oh my zsh install not after it - the way you've got it here it won't load the aliases for ```tzsh``` (ditto for the other dockerfiles)

Comment (markmetcalfe) [reply php/php72/Dockerfile:142]: Or alternatively you can duplicate this line, but pipe the output to ```.zshrc```
 ```suggestion
RUN echo 'for f in ~/custom_shell/*.sh; do [[ -e "$f" ]] && source "$f"; done;' >> ~/.bashrc
RUN echo 'for f in ~/custom_shell/*.sh; do [[ -e "$f" ]] && source "$f"; done;' >> ~/.zshrc
```

**Review by markmetcalfe — CHANGES_REQUESTED** [review]

**Pull request closed without merging**